src/compiler/parser.ts

```diff
@@ -6089,6 +6089,7 @@ namespace ts {
             const referencedFiles: FileReference[] = [];
             const typeReferenceDirectives: FileReference[] = [];
             const amdDependencies: { path: string; name: string }[] = [];
+            let pragmas: { pragma: string, value: string | undefined }[] = [];

```

Comment:
if we gonna add a new concept, then we should consolidate the other ones under pargma.. and we should allow `///` as well as multi-line comments.

I would also have helper functions for getting all of these like amdDependecies, amdModuleNAme, checkJs, etc..

Reply:
Sure! I wanted to consolidate them, anyway, just didn't think this PR was appropriate.

Reply:
We need to ensure this works well when doing incremental parsing as well and need test for the same.